### PR TITLE
Track the number of MySQL connections

### LINKED ISSUE #9013: Implement system.metrics.MySQLConnection
Based on review https://github.com/ClickHouse/ClickHouse/tree/master/dbms/programs/server/MySQLHandler.h
currently, MySQL connections don't show in any `system.metrics` gauges and also don't include on `system.metrics.TCPConnection`

**Use case**
Show MySQL connections as separate metrics should improve monitoring cases about current connections

**Describe the solution you'd like**
Add MySQLConnection metric to https://github.com/ClickHouse/ClickHouse/tree/master/dbms/src/Common/CurrentMetrics.cpp

Implements increment this metric on 
https://github.com/ClickHouse/ClickHouse/tree/master/dbms/programs/server/MySQLHandler.h

### PR DESCRIPTION
**Pull request by Slach**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add a system metric tracking the number of client connections using MySQL protocol (#9013).

fix https://github.com/ClickHouse/ClickHouse/issues/9013

### CODE REVIEW

**Comment by Slach** [comment]
@yurriy could you also look at this pull request?

**Review by akuzm — APPROVED** [review]

File: dbms/src/Common/CurrentMetrics.cpp [hunk 1]
```diff
@@ -15,7 +15,8 @@
     M(DiskSpaceReservedForMerge, "Disk space reserved for currently running background merges. It is slightly more than the total size of currently merging parts.") \
     M(DistributedSend, "Number of connections to remote servers sending data that was INSERTed into Distributed tables. Both synchronous and asynchronous mode.") \
     M(QueryPreempted, "Number of queries that are stopped and waiting due to 'priority' setting.") \
-    M(TCPConnection, "Number of connections to TCP server (clients with native interface)") \
+    M(TCPConnection, "Number of connections to TCP server (clients with native interface), also included server-server distributed query connections") \
+    M(MySQLConnection, "Number of connections of clients over MySQL wire protocol") \
```
Comment (yurriy) [inline dbms/src/Common/CurrentMetrics.cpp]: Maybe "Number of connections to ClickHouse's MySQL compatibility server" sounds better

Comment (abyss7) [reply dbms/src/Common/CurrentMetrics.cpp]: "Number of client connections using MySQL wire protocol"

**Review by abyss7 — APPROVED** [review]

File: dbms/src/Common/CurrentMetrics.cpp (hunk 1 above)
Comment (Slach) [reply dbms/src/Common/CurrentMetrics.cpp]: fixed thanks @abyss7

**Review by yurriy — APPROVED** [review]

**Pull request merged**